src/task_types/resmoke_tasks.rs

```diff
@@ -375,6 +395,33 @@ impl GenResmokeTaskServiceImpl {
         Ok(sub_suites)
     }
 
+    /// Get the list of tests belonging to the suite being generated.
+    ///
+    /// # Arguments
+    ///
+    /// * `params` - Parameters about the suite being split.
+    ///
+    /// # Returns
+    ///
+    /// List of tests belonging to suite being split.
+    fn get_test_list(&self, params: &ResmokeGenParams) -> Result<Vec<String>> {
+        let mut test_list: Vec<String> = self
+            .test_discovery
+            .discover_tests(&params.suite_name)?
+            .into_iter()
+            .filter(|s| self.fs_service.file_exists(s))
+            .collect();
+
+        if !params.is_enterprise {
+            test_list = test_list
+                .into_iter()
+                .filter(|s| !s.starts_with(&self.config.enterprise_prefix))
```

Comment:
Enterprise module prefix `src/mongo/db/modules` seems too broad and not enough specific to enterprise module itself.
https://github.com/mongodb/mongo/blob/7371fa1eb1008f2be98a1acdaa818c4ac82e9fa8/etc/evergreen_yml_components/definitions.yml#L73
```
modules:
- name: enterprise
  repo: git@github.com:10gen/mongo-enterprise-modules.git
  prefix: src/mongo/db/modules
  branch: master
```
I think it should be appended with `enterprise`, so that we filter out tests that are from `src/mongo/db/modules/enterprise` directory.

Reply:
updated.